### PR TITLE
CB-12897 use existing vpc and subnet during distrox provision with AW…

### PR DESCRIPTION
**Pull request by topolyai5**
…S_NATIVE variant

See detailed description in the commit message.

### CODE REVIEW

**Review by doktoric — APPROVED** [review]

File: cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/AwsVolumeResourceBuilder.java [hunk 1]
```diff
@@ -320,11 +321,20 @@ public ResourceType resourceType() {
...
-        ResourceStatus volumeSetStatus = getResourceStatus(result);
+        AtomicReference<ResourceStatus> volumeSetStatus = new AtomicReference<>();
+        try {
+            DescribeVolumesResult result = client.describeVolumes(describeVolumesRequest);
+            volumeSetStatus.set(getResourceStatus(result));
+        } catch (AmazonEC2Exception e) {
+            if (!"InvalidVolume.NotFound".equals(e.getErrorCode())) {
+                throw e;
+            }
+            LOGGER.debug("Volume set to deleted because: {}", e.getMessage());
```
Comment (biharitomi) [inline cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/AwsVolumeResourceBuilder.java:332]: More likely -> 'The volume doesn't need to be deleted as it does not exist on the provider side' and I would prefer info level..

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/client/AmazonEc2Client.java [hunk 2]
```diff
@@ -211,4 +215,11 @@ public AuthorizeSecurityGroupIngressResult addIngress(AuthorizeSecurityGroupIngr
     public AuthorizeSecurityGroupEgressResult addEgress(AuthorizeSecurityGroupEgressRequest reguest) {
         return client.authorizeSecurityGroupEgress(reguest);
     }
+
+    // WIP
```
Comment (biharitomi) [inline cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/client/AmazonEc2Client.java:219]: Do we need this comment?

File: cloud-aws-native/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/network/AwsStackUtil.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+public class AwsStackUtil {
+
+    public static final String VPC_ID = "vpcId";
+
+    public static final String SUBNET_ID = "subnetId";
+
+    public static final String NO_PUBLIC_IP = "noPublicIp";
+
+    public boolean isExistingNetwork(AmazonEc2Client amazonEc2Client, Network network) {
+        DescribeVpcsResult result = amazonEc2Client.describeVpcs(new DescribeVpcsRequest().withVpcIds(network.getStringParameter(VPC_ID)));
```
Comment (biharitomi) [inline cloud-aws-native/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/network/AwsStackUtil.java:20]: I think exception handling would be nice here as AWS throw an exception with some kind of `NotFound` error code when a resource does not exist in most of the cases.

**Review by biharitomi — CHANGES_REQUESTED** [review]

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/client/AmazonEc2Client.java (hunk 2 above)
Comment (topolyai5) [reply cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/client/AmazonEc2Client.java:219]: Yes, we need it. The vpc creation will be finished in the next task, until this I don't want to delete this code snippet.

File: cloud-aws-native/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/network/AwsStackUtil.java (hunk 3 above)
Comment (topolyai5) [reply cloud-aws-native/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/network/AwsStackUtil.java]: done

File: cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/AwsVolumeResourceBuilder.java (hunk 1 above)
Comment (topolyai5) [reply cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/AwsVolumeResourceBuilder.java]: done

**Review by biharitomi — APPROVED** [review]
LGTM

**Pull request merged**